aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -34,6 +40,40 @@ impl NativeTransactionContext {
     }
 }
 
+/***************************************************************************************************
+ * native fun create_guid
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Clone, Debug)]
+pub struct CreateGUIDGasParameters {
+    pub base: InternalGas,
+}
+
+fn native_create_guid(
+    gas_params: &CreateGUIDGasParameters,
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(gas_params.base)?;
+
+    let mut transaction_context = context
+        .extensions_mut()
+        .get_mut::<NativeTransactionContext>();
+    transaction_context.guid_counter += 1;
+    let mut hash_arg = Vec::new();
+    hash_arg.extend(transaction_context.txn_hash.clone());
```

Comment:
you need to append this as well:

  const OBJECT_FROM_TRANSACTION_GUID_ADDRESS_SCHEME: u8 = 0xFB;

to make it unique among the ways guids are created

Reply:
Done.